src/test/java/uk/gov/pay/webhooks/deliveryqueue/managed/WebhookMessagePollingServiceTest.java

```diff
@@ -0,0 +1,155 @@
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class WebhookMessagePollingServiceTest {
+
+    private final DAOTestExtension database = DAOTestExtension.newBuilder()
+            .addEntityClass(EventTypeEntity.class)
+            .addEntityClass(WebhookMessageEntity.class)
+            .addEntityClass(WebhookEntity.class)
+            .addEntityClass(WebhookDeliveryQueueEntity.class)
+            .build();
+
+    private InstantSource instantSource;
+    private WebhookDeliveryQueueDao webhookDeliveryQueueDao;
+    private WebhookMessageDao webhookMessageDao;
+    private WebhookDao webhookDao;
+    private WebhookMessagePollingService webhookMessagePollingService;
+    private WebhookMessageSender webhookMessageSenderMock;
+    private SendAttempter sendAttempter;
+    @Mock
+    private HttpResponse response;
+
+    @BeforeEach
+    void setUp() throws IOException, InvalidKeyException, InterruptedException {
+        var environment = mock(Environment.class);
+
+        when(environment.metrics()).thenReturn(mock(MetricRegistry.class));
+        when(response.statusCode()).thenReturn(200);
+        webhookMessageSenderMock = mock(WebhookMessageSender.class);
+        when(webhookMessageSenderMock.sendWebhookMessage(any(WebhookMessageEntity.class))).thenReturn(response);
+        instantSource = InstantSource.fixed(Instant.now());
+        webhookMessageDao = new WebhookMessageDao(database.getSessionFactory());
+        webhookDao = new WebhookDao(database.getSessionFactory());
+        webhookDeliveryQueueDao = new WebhookDeliveryQueueDao(database.getSessionFactory(), instantSource);
+        sendAttempter = new SendAttempter(webhookDeliveryQueueDao, instantSource, webhookMessageSenderMock, environment);
+        webhookMessagePollingService = new UnitOfWorkAwareProxyFactory("default", database.getSessionFactory())
+                .create(
+                        WebhookMessagePollingService.class,
+                        new Class[] { WebhookDeliveryQueueDao.class, SendAttempter.class },
+                        new Object[] { webhookDeliveryQueueDao, sendAttempter }
+                );
+    }
+
+    @Test
+    public void shouldPollWithNoActionsOnEmptyQueue() {
+        database.inTransaction(() -> {
+            webhookMessagePollingService.pollWebhookMessageQueue();
+            verifyNoInteractions(webhookMessageSenderMock);
+        });
+    }
+
+    @Test
+    public void shouldEmitSingleValidQueueItem() {
+        setupOrderedDeliveryQueueWith(List.of("first-external-id"));
+        database.inTransaction(() -> {
+            webhookMessagePollingService.pollWebhookMessageQueue();
+            try {
+                verify(webhookMessageSenderMock).sendWebhookMessage(argThat((webhookMessage -> webhookMessage.getExternalId().equals("first-external-id"))));
+            } catch (IOException | InvalidKeyException | InterruptedException ignored) { }
```

Comment:
It's being executed in a lambda expression (call to db inTransaction) because of the nature of the unit test (hibernate doesn't other have an existing session as the apps not running or managing it), so unfortunately these errors don't belong to the test method signature -- it was like this before but now that we're no longer manually managing the hibernate transactions the lambda expression is needed here

Reply:
That makes sense. If we do not expect these exceptions to happen, stick a `fail()` call inside the catch block so that if they ever do happen, we immediately fail the test.